src/NaluParsing.C

```diff
@@ -229,6 +229,30 @@ namespace sierra
       abltopBC.theBcType_ = ABLTOP_BC;
       const YAML::Node& abltopUserData = node["abltop_user_data"];
       abltopBC.userData_ = abltopUserData.as<ABLTopUserData>();
+      if (node["symmetry_user_data"]) {
```

Comment:
This looks like it was copied from the symmetry bc. Consider extracting this as a function so we don't have to maintain the same syntax in two places?

Reply:
Do you have a suggestion for how to do this? I tried to avoid this by using e.g. `get_if_present_no_default` or `get_if_present` but kept running into errors, which Jeremy, Marc, and I couldn't get around.

Reply:
Something like this?

```c++
// function somewhere else:
SymmetryUserData get_symmetry_data(const YAML::Node& node)
{
        const YAML::Node& symmetryUserData = node["symmetry_user_data"];
        auto data = symmetryUserData.as<SymmetryUserData>();
        if (symmetryUserData["symmetry_type"]) {
          const std::string symmType = symmetryUserData["symmetry_type"].as<std::string>();
          if(symmType == "generalized_weak" || symmType == "default"){
            // do nothing already set, but keep for parse checking
          }
          else if(symmType == "x_direction_strong"){
            data_.symmType_ = SymmetryUserData::SymmetryTypes::X_DIR_STRONG;
          }
          else if(symmType == "y_direction_strong"){
            data_.symmType_ = SymmetryUserData::SymmetryTypes::Y_DIR_STRONG;
          }
          else if(symmType == "z_direction_strong"){
            data_.symmType_ = SymmetryUserData::SymmetryTypes::Z_DIR_STRONG;
          }
          else{
            throw std::runtime_error(
              "Unrecognized value for symmetry_type: " + symmType
              );
          }
        } 
  return data;
} 

// your code now
      if (node["symmetry_user_data"]) {
         abltopBC.symmetryUserData_ = get_symmetry_data(node);
      }      
```

Reply:
Ah, okay. I misunderstood. I thought you were objecting to me copying `node["symmetry_user_data"]` instead of using a function like `get_if_present_no_default`.

I made the suggested change.